### PR TITLE
[API remove] remove variantImpl & variantwrapper related class/interfaces

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
Signed-off-by: xuejun <xuejun.zhai@intel.com>

### Details:
 - remove class variantImpl 
 - remove class variantwrapper 
 - remove the classes related APIs

### Tickets:
 - 100091

### CODE REVIEW

File: src/common/low_precision_transformations/src/rt_info/quantization_mode_attribute.cpp [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 #include "low_precision/rt_info/quantization_mode_attribute.hpp"
 #include <assert.h>
+#include <sstream>
```
Comment (riverlijunjie) [inline src/common/low_precision_transformations/src/rt_info/quantization_mode_attribute.cpp:8]: Why add this header file?

File: src/common/transformations/src/transformations/fix_rt_info.cpp [hunk 2]
```diff
@@ -23,17 +22,12 @@ bool ov::pass::FixRtInfo::run_on_model(const std::shared_ptr<ngraph::Function>&
...
             }
         }
-        OPENVINO_SUPPRESS_DEPRECATED_START
+        
         auto& rt_info = node->get_rt_info();
         {
             auto it_info = rt_info.find("PrimitivesPriority");
             if (it_info != rt_info.end()) {
-                if (it_info->second.is<std::shared_ptr<ngraph::VariantWrapper<std::string>>>()) {
-                    rt_info.emplace(
```
Comment (riverlijunjie) [inline src/common/transformations/src/transformations/fix_rt_info.cpp:32]: Is there any problem if delete this branch?

**Comment by ilyachur** [comment]
Please fix first of all external components and code style

File: src/common/low_precision_transformations/src/rt_info/quantization_mode_attribute.cpp (hunk 1 above)
Comment (zhaixuejun1993) [reply src/common/low_precision_transformations/src/rt_info/quantization_mode_attribute.cpp:8]: removed the header file "#include <ngraph/variant.hpp>" for "quantization_mode_attribute.hpp" caused line 14 "std::stringstream ss;" undefined, so add this header file.

Comment (zhaixuejun1993) [reply src/common/low_precision_transformations/src/rt_info/quantization_mode_attribute.cpp:8]: > Why add this header file?

removed the header file "#include <ngraph/variant.hpp>" for "quantization_mode_attribute.hpp" caused line 14 "std::stringstream ss;" undefined, so add this header file.

File: src/common/transformations/src/transformations/fix_rt_info.cpp (hunk 2 above)
Comment (zhaixuejun1993) [reply src/common/transformations/src/transformations/fix_rt_info.cpp:32]: > Is there any problem if delete this branch?

Form code perspective, this branch operate with "VariantWrapper" seems like can be removed directly. @ilyachur  how do u think?

**Comment by zhaixuejun1993** [comment]
> Please fix first of all external components and code style

Created the PR for arm plugin, pls help review.
https://github.com/openvinotoolkit/openvino_contrib/pull/592

**Comment by ilya-lavrenov** [comment]
> > Please fix first of all external components and code style
> 
> Created the PR for arm plugin, pls help review. [openvinotoolkit/openvino_contrib#592](https://github.com/openvinotoolkit/openvino_contrib/pull/592)

merged, thanks

File: src/common/transformations/src/transformations/fix_rt_info.cpp (hunk 2 above)
Comment (ilyachur) [reply src/common/transformations/src/transformations/fix_rt_info.cpp:32]: @itikhono Could you comment?

File: src/bindings/python/src/compatibility/pyngraph/rt_map.cpp [hunk 3]
```diff
@@ -36,6 +35,6 @@ void regclass_pyngraph_PyRTMap(py::module m) {
         m[k] = v;
     });
     py_map.def("__getitem__", [](PyRTMap& m, const std::string& k) {
-        return m.at(k).as<std::shared_ptr<ngraph::Variant>>();
+        return m.at(k).as<std::shared_ptr<ov::RuntimeAttribute>>();
```
Comment (ilya-lavrenov) [inline src/bindings/python/src/compatibility/pyngraph/rt_map.cpp:38]: @jiwaszki is it a safe change?

File: src/common/transformations/src/transformations/fix_rt_info.cpp [hunk 4]
```diff
@@ -23,17 +22,11 @@ bool ov::pass::FixRtInfo::run_on_model(const std::shared_ptr<ngraph::Function>&
                 run_on_model(sub_graph);
```
Comment (zhaixuejun1993) [inline src/common/transformations/src/transformations/fix_rt_info.cpp:22]: @ilya-lavrenov u mean this func can be remove? Caused the shared_ptr had been removed, so no need to  extract runtime attributes from shared pointer any more?

Comment (ilya-lavrenov) [reply src/common/transformations/src/transformations/fix_rt_info.cpp:22]: we discussed with @ilyachur and decided to remove transformation separately. 
So, let's keep it for now

**Review by ilyachur — APPROVED** [review]

**Pull request merged**